crates/dekoder/src/lib.rs

```diff
@@ -0,0 +1,172 @@
+//! eko output interface.
+use float_cmp::approx_eq;
+use ndarray::Array4;
+use std::collections::HashMap;
+use std::fs::remove_dir_all;
+use std::fs::File;
+use std::io::BufWriter;
+use std::path::PathBuf;
+use thiserror::Error;
+use yaml_rust2::Yaml;
+
+mod inventory;
+
+/// The EKO errors.
+#[derive(Error, Debug)]
+pub enum EKOError {
+    #[error("No working directory")]
+    NoWorkingDir,
+    #[error("I/O error")]
+    IOError(#[from] std::io::Error),
+    #[error("Loading operator from `{0}` failed")]
+    OperatorLoadError(PathBuf),
+    #[error("Failed to read key(s) `{0}`")]
+    KeyError(String),
+}
+
+/// My result type has always my errros.
+type Result<T> = std::result::Result<T, EKOError>;
+
+/// Default value for the float comparison inside `EvolutionPoint`.
+const EP_CMP_ULPS: i64 = 64;
+
+/// A reference point in the evolution atlas.
+pub struct EvolutionPoint {
+    /// Evolution scale.
+    pub scale: f64,
+    /// Number of flavors
+    pub nf: i64,
+}
+
+/// 4D evolution operator.
+pub struct Operator {
+    pub op: Option<Array4<f64>>,
+}
+
+impl Operator {
+    /// Empty initializer.
+    pub fn new() -> Self {
+        Self { op: None }
+    }
+}
+
+impl Default for Operator {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+impl TryFrom<&Yaml> for EvolutionPoint {
+    type Error = EKOError;
+
+    /// Load from yaml.
+    fn try_from(yml: &Yaml) -> Result<Self> {
+        // work around float representation
+        let scale = yml["scale"].as_f64();
+        let scale = if scale.is_some() {
+            scale.ok_or(EKOError::KeyError(
+                "because failed to read scale as float".to_owned(),
+            ))?
+        } else {
+            yml["scale"].as_i64().ok_or(EKOError::KeyError(
+                "because failed to read scale as float from int".to_owned(),
+            ))? as f64
+        };
+        let nf = yml["nf"]
+            .as_i64()
+            .ok_or(EKOError::KeyError("because failed to read nf".to_owned()))?;
+        Ok(Self { scale, nf })
+    }
+}
+
+impl PartialEq for EvolutionPoint {
+    /// (Protected) comparator.
+    fn eq(&self, other: &Self) -> bool {
+        self.nf == other.nf && approx_eq!(f64, self.scale, other.scale, ulps = EP_CMP_ULPS)
+    }
+}
+
+impl Eq for EvolutionPoint {}
+
+/// EKO output
+pub struct EKO {
+    /// Working directory
+    path: PathBuf,
+    /// final operators
+    operators: inventory::Inventory<EvolutionPoint>,
+}
+
+/// Operators directory.
+const DIR_OPERATORS: &str = "operators/";
+
+impl EKO {
+    /// Check our working directory is safe.
+    fn check(&self) -> Result<()> {
+        let path_exists = self.path.try_exists().is_ok_and(|x| x);
```

Comment:
I believe this is equivalent to:
```suggestion
        let path_exists = self.path.exists();
```
> #### See Also
> This is a convenience function that coerces errors to false. If you want to check errors, call [Path::try_exists](https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists).
> https://doc.rust-lang.org/std/path/struct.Path.html#method.exists

Otherwise, if you want to receive the errors as errors, it could be better to use:
```suggestion
        let path_exists = self.path.try_exists()?;
```
since:
> [Path::exists()](https://doc.rust-lang.org/std/path/struct.Path.html#method.exists) only checks whether or not a path was both found and readable. By contrast, try_exists will return Ok(true) or Ok(false), respectively, if the path was verified to exist or not exist.
> https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists

Reply:
I opted for this way, because it is precisely suggested there

> Warning: this method may be error-prone, consider using [try_exists()](https://doc.rust-lang.org/std/path/struct.Path.html#method.try_exists) instead! It also has a risk of introducing time-of-check to time-of-use (TOCTOU) bugs.

and since we may eventually do want to run in parallel, this could be an issue

Reply:
> Note that while this avoids some pitfalls of the exists() method, it still can not prevent time-of-check to time-of-use (TOCTOU) bugs. You should only use it in scenarios where those bugs are not an issue.

Reply:
ok  - done in 5be3f186 (with the compressed version from below)